packages/block-library/src/cover/edit/inspector-controls.js

```diff
@@ -112,7 +117,38 @@ export default function CoverInspectorControls( {
 		overlayColor,
 	} = currentSettings;
 
+	const sizeSlug = attributes.sizeSlug || DEFAULT_MEDIA_SIZE_SLUG;
+
 	const { gradientValue, setGradient } = __experimentalUseGradient();
+	const { getSettings } = useSelect( blockEditorStore );
+
+	const imageSizes = getSettings().imageSizes;
```

Comment:
It appears as though there are a large number of places in the codebase where the presence of `imageSizes` in the settings is assumed. This could still be a touch more robust in the face of future changes if we fell back to an empty array if it isn't available. Up to you if you'd like to tweak it before landing this.

Reply:
If `imageSizes` is set to empty, `ResolutionTool` wont be visible. We could set fallback value for this, but if user have set imageSizes setting to empty array, should we fallback it to default value again?

Also, We could create hook to set default value if imageSizes setting is empty.

Reply:
If it is null or undefined, the filter and map call below would throw an error.

Reply:
I see.

Reply:
I have added optional chaining operator to avoid errors. Thanks for the feedback.